src/hook-check.ts

```diff
@@ -0,0 +1,88 @@
+import { Application } from './application'
+import { GitHubAPI } from './github'
+
+const MESSAGES: {
+  [key: string]: string[]
+} = {
+  DISABLE_WEBHOOK_CHECK: ['You can disable webhook checking by setting the environment variable DISABLE_WEBHOOK_CHECK to "true".'],
+  ISSUE_REPORT: [
+    'If this error persists, you can raise a bug report at:',
+    '  - https://github.com/probot/probot/issues'
+  ]
+}
+
+let appMetadata: ReturnType<GitHubAPI['apps']['getAuthenticated']> | null = null
+
+let hasDisplayedFeatureDisabledWarning = false
+function disabledHookCheckWarning (app: Application) {
+  if (!hasDisplayedFeatureDisabledWarning) {
+    app.log.debug('DISABLE_WEBHOOK_CHECK is enabled in your environment. You will not be warned if your Probot app is attempting to listen to an event it is not subscribed to.')
+  }
+
+  hasDisplayedFeatureDisabledWarning = true
+}
+
+async function hookCheck (app: Application, eventName: string) {
+  if ((process.env.DISABLE_WEBHOOK_CHECK && process.env.DISABLE_WEBHOOK_CHECK.toLowerCase() === 'true') || process.env.NODE_ENV === 'production') {
```

Comment:
I would suggest we also don't do the check if `NODE_ENV` is set to `production`

Reply:
Do we want this to run in production? It will make one call to the `/app` endpoint, and even though this doesn't appear to count against an application's rate limit, in a serverless environment the call will be made every single time the serverless application is invoked.

Reply:
What I meant to say is that I'd suggest to _not_ run this in production, only in development

Reply:
Alright, this will only run in environments where `NODE_ENV` is `development`.

Reply:
I don't think that `process.env.NODE_ENV` is set to `development` by default. I think I'd do the reverse, and check if `NODE_ENV` is set to `production` which is more common, I think

Reply:
Okay, the feature will not run production.

https://github.com/probot/probot/pull/1078/files#diff-30b8bc5f017c2a2015bdf18c027e5801R91-R98